File: Medium/2026-06-27-3020-Find-the-Maximum-Number-of-Elements-in-Subset/solution.js
var maximumSubsetSize = function(nums) {
    const count = new Map();
    
    for (const num of nums) {
        count.set(num, (count.get(num) || 0) + 1);
    }
    
    let maxSize = 0;
    
    for (const [num, freq] of count) {
        let currentSize = freq;
        let power = num;
        
        while (count.has(power * 2)) {
            currentSize += count.get(power * 2);
            power *= 2;
        }
        
        maxSize = Math.max(maxSize, currentSize);
    }
    
    return maxSize;
};

File: Medium/2026-06-27-3020-Find-the-Maximum-Number-of-Elements-in-Subset/solution.java
import java.util.HashMap;
import java.util.Map;

public class Solution {
    public int findMaximumSubsetSize(int[] nums) {
        Map<Integer, Integer> countMap = new HashMap<>();
        for (int num : nums) {
            countMap.put(num, countMap.getOrDefault(num, 0) + 1);
        }

        int maxSize = 0;
        for (int num : countMap.keySet()) {
            int size = countMap.get(num);
            int power = num * 2;
            while (countMap.containsKey(power)) {
                size += countMap.get(power);
                power *= 2;
            }
            maxSize = Math.max(maxSize, size);
        }

        return maxSize;
    }
}